optaplanner-examples/src/main/java/org/optaplanner/examples/batchscheduling/domain/Segment.java

```diff
@@ -0,0 +1,77 @@
+package org.optaplanner.examples.batchscheduling.domain;
+
+import org.optaplanner.examples.common.domain.AbstractPersistable;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+
+@XStreamAlias("PipeSegment")
+public class Segment extends AbstractPersistable {
+
+    private Batch batch;
+    private RoutePath routePath;
+    private String name;
+    private Integer sequence;
+
+    //flowRate is in m3/minute
+    private float flowRate;
+
+    private float length;
```

Comment:
If you use this anywhere in your scoring function, you could be badly hurt. Float/double precision has been shown to affect OptaPlanner adversely.

In that case, I suggest you use BigDecimal or - if you need max performance - to use fixed-point arithmetic.
But if you don't use this in a scoring function, you should be fine.